x/protocolpool/keeper/keeper.go

```diff
@@ -29,12 +38,24 @@ func NewKeeper(cdc codec.BinaryCodec, storeService storetypes.KVStoreService,
 	if addr := ak.GetModuleAddress(types.ModuleName); addr == nil {
 		panic(fmt.Sprintf("%s module account has not been set", types.ModuleName))
 	}
-	return Keeper{
-		storeService: storeService,
-		authKeeper:   ak,
-		bankKeeper:   bk,
-		authority:    authority,
+	sb := collections.NewSchemaBuilder(storeService)
+
+	keeper := Keeper{
+		storeService:   storeService,
+		authKeeper:     ak,
+		bankKeeper:     bk,
+		cdc:            cdc,
+		authority:      authority,
+		BudgetProposal: collections.NewMap(sb, types.BudgetKey, "budget", sdk.AccAddressKey, codec.CollValue[types.MsgSubmitBudgetProposal](cdc)),
```

Comment:
I think we should create a type called `Budget` and use that.
Then the keeper creates a `Budget` and the msg server does the MsgSubmitBudgetProposal -> BudgetProposal.
A bit like we do for gov proposals.